File: deploy/server/scripts/resync-deal-amo.js
"use strict";

const { getAccessToken, amoGetAll } = require("/opt/itmen-pipeline/api/src/amo-client");
const { findOne } = require("/opt/itmen-pipeline/api/src/pb-client");
const { syncLeadNotesAndTasks } = require("/opt/itmen-pipeline/api/src/amo-lead-sync");

async function main() {
  const dealId = process.argv[2] || "D-1009";
  const row = await findOne("deals", `deal_id="${String(dealId).replace(/"/g, '\\"')}"`);
  if (!row?.amo_id) throw new Error(`No amo_id for ${dealId}`);
  const token = await getAccessToken();
  const leads = await amoGetAll("/api/v4/leads", token, { "filter[id]": row.amo_id, with: "contacts" });
  const lead = leads[0];
  if (!lead) throw new Error(`Lead ${row.amo_id} not found in Amo`);
  await syncLeadNotesAndTasks({ lead, token, dealId, pbId: row.id });
  console.log("resynced", dealId, "amo", row.amo_id);
}

main().catch(e => { console.error(e); process.exit(1); });

File: deploy/server/scripts/resync-deal-amo.py
#!/usr/bin/env python3
import json
import sys
import urllib.request

DEAL_ID = sys.argv[1] if len(sys.argv) > 1 else "D-1009"
API = "http://127.0.0.1:3010"


def load_env(path):
    env = {}
    for line in open(path, encoding="utf-8"):
        line = line.strip()
        if not line or line.startswith("#") or "=" not in line:
            continue
        k, v = line.split("=", 1)
        env[k.strip()] = v.strip().strip('"')
    return env


def main():
    env = load_env("/opt/itmen-pipeline/.env")
    login = urllib.request.Request(
        f"{API}/api/auth/login",
        data=json.dumps({
            "email": env.get("ADMIN_EMAIL") or env.get("PB_ADMIN_EMAIL"),
            "password": env.get("ADMIN_PASSWORD") or env.get("PB_ADMIN_PASSWORD"),
        }).encode(),
        headers={"Content-Type": "application/json"},
        method="POST",
    )
    session = json.load(urllib.request.urlopen(login))
    token = session.get("token") or session.get("accessToken")
    if not token and session.get("user"):
        token = session.get("token")
    headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json"}
    req = urllib.request.Request(
        f"{API}/api/deals/{DEAL_ID}/amo-resync",
        data=b"{}",
        headers=headers,
        method="POST",
    )
    print(json.dumps(json.load(urllib.request.urlopen(req)), ensure_ascii=False, indent=2))


if __name__ == "__main__":
    main()
